File: mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

int get_height(void);
void print_grid(int height);

int main(void)
{

    // gets height of grid
    int n = get_height();

    // prints grid
    print_grid(n);

}

















int get_height(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n > 8 || n < 1);
    return n;
}

void print_grid(int height)

{
    for (int i = 0; i < height; i++)
        // helpful to think of "i" as the line you are on... the loop will iterate a certain amount of times depeneding on height variable

    {
        for (int j = i + 2; j <= height; j++) // this loop adds spaces to the left column
        {
            printf(" ");
        }

        for (int k = 0; k <= i; k++) // the following loops adds the hashes and space between the hashes
        {
            printf("#");
        }

        printf("  ");

        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }

        printf("\n");

    }
}

File: sentimental-mario-more/mario.py
import cs50


def main():

    art = getheight()

    print_grid(art)


def getheight():

    while True:
        height = cs50.get_int("Height: ")
        if height < 9 and height > 0:
            break

    return height


def print_grid(height):

    for i in range(height):
        print(" " * (height - (i + 1)), end="")
        print("#" * (i + 1), end="")
        print("  ", end="")
        print("#" * (i + 1), end="")
        print("")


if __name__ == "__main__":
    main()